apps/cmp.c

```diff
@@ -809,7 +809,11 @@ static int load_certs_preliminary(const char *file, STACK_OF(X509) **certs,
     int ret = 0;
 
     if (format == FORMAT_PKCS12) {
-        BIO *bio = bio_open_default(file, 'r', format);
+        FILE *fp = fopen(file, "r");
+        BIO *bio = NULL;
+
+        if (fp != NULL)
+            bio = BIO_new_fp(fp, format);
```

Comment:
Yes, such a change is needed.
The question would have not arisen if the review of #4930 had been finalized and the changes merged, because then the whole `load_certs_preliminary()` function would be gone.